modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/workflow/WorkflowExecutorFactory.java

```diff
@@ -0,0 +1,132 @@
+                    + "dynamically";
+            log.debug(message);
+        }
+
+        workflowExecutorMap = new ConcurrentHashMap<WorkflowConstant.WorkflowType, WorkflowExecutor>();
+        String executorClass;
+        WorkflowExecutor workflowExecutor;
+        try {
+            executorClass = AppFactoryUtil.getAppfactoryConfiguration()
+                    .getFirstProperty(WorkflowConstant.TENANT_CREATION);
+        } catch (AppFactoryException e) {
+            String message = "Unable to read executor class from the appfactory.xml configuration";
+            throw new AppFactoryException(message, e);
+        }
+
+        if (StringUtils.isBlank(executorClass)) {
+            executorClass = WorkflowConstant.TENANT_CREATION_DEFAULT_WORKFLOW_CLASS;
+            String message = "The tenant creation executor class is not define in appfactory.xml because tenant "
+                    + "creation default workflow is " + "executing. The tenant creation default workflow class is : "
+                    + executorClass;
+            log.warn(message);
+        }
+
+        try {
+            Class tenantCreationExecutorClass = WorkflowExecutorFactory.class.getClassLoader().loadClass(executorClass);
+            workflowExecutor = (WorkflowExecutor) tenantCreationExecutorClass.newInstance();
+            workflowExecutorMap.put(WorkflowConstant.WorkflowType.TENANT_CREATION, workflowExecutor);
+        } catch (ClassNotFoundException e) {
+            String message = "Unable to find configured class from the source code : class name : " + executorClass;
+            throw new AppFactoryException(message, e);
+        } catch (InstantiationException e) {
+            String message = "Can not initiate object of the class : class name : " + executorClass;
+            throw new AppFactoryException(message, e);
+        } catch (IllegalAccessException e) {
+            String message = "Can not access the class to create new instance : class name : " + executorClass;
+            throw new AppFactoryException(message, e);
+        }
+    }
+
+    /**
+     * Return workflow object by given workflow type
+     *
+     * @param workflowType type of the workflow
+     * @return workflow object
+     */
+    public WorkflowExecutor getWorkflowExecutor(WorkflowConstant.WorkflowType workflowType) {
+        return workflowExecutorMap.get(workflowType);
+    }
+
+    /**
+     * Create workflow data transfer object by given workflow type
+     *
+     * @param workflowType type of the workflow
+     * @return workflow DTO object
+     */
+    public WorkflowDTO createWorkflowDTO(WorkflowConstant.WorkflowType workflowType) {
+        WorkflowDTO workflowDTO;
+
+        switch (workflowType) {
+        case TENANT_CREATION:
+            workflowDTO = new TenantCreationWorkflowDTO();
+            break;
+
+        default:
+            String message = "The type of workflow not support : workflow type : " + workflowType;
```

Comment:
Format the log message in more user friendly manner. No need of extra " : " middle of the sentence